Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,453 @@
+                                            , DestinationIP has ":", "IPv6"
+                                            , ""
+                                            )
+    , NetworkProtocol = toupper(trim_end(@"/.*", Protocol))
+    , NetworkApplicationProtocol = tostring(toupper(trim_start(@".*/", Protocol)))
+    , sosLogMsgID = DeviceEventClassID
+  | project-rename
+    DstMacAddr = DestinationMACAddress
+    , SrcMacAddr = SourceMACAddress
+    , DstIpAddr = DestinationIP
+    , SrcIpAddr = SourceIP
+    , DstPortNumber = DestinationPort
+    , SrcPortNumber = SourcePort
+    , EventMessage = Activity
+    , sosEventMessageDetail = Message
+    , EventProductVersion = DeviceVersion
+    , sosSerialNumber = Computer
+    , DvcOutboundInterface = DeviceOutboundInterface
+    , DvcInboundInterface = DeviceInboundInterface
+    , sosApplicationID = ApplicationProtocol // Application ID number (when Flow Reporting is enabled).
+    , sosCFSFullString = Reason // CFS Category ID and Name
+    , sosRuleDetails = DeviceCustomString1 // Rule ID. Identify a policy or rule associated with an event.
+    , sosSourceVPNPolicyName = DeviceCustomString2 // Displays the source VPN policy name associated with the event.
+    , sosDestinationVPNPolicyName = DeviceCustomString3 // Displays the destination VPN policy name associated with the event.
+    , sosInterfaceStatistics = DeviceCustomString4 // Displays interface statistics.
+    , sosSoleraNPCSURL = DeviceCustomString5 // URL. Applicable only when NPCS Solera is enabled. Displays the URL of an NPCS object.
+    , sosLogMsgNote = DeviceCustomString6 // "Note" field. Additional information that is application-dependent.
+    , sosNATSourceIPAddress = DeviceCustomString1Label // NAT'ed source IP4/IPv6 address.
+    , sosNATDestinationIPAddress = DeviceCustomString2Label // NAT'ed destination IPv4/IPv6 address.
+    , sosSourceZone = DeviceCustomString3Label // Source Zone on Gen7. Src Zone Type on Gen6.
+    , sosDestinationZone = DeviceCustomString4Label // Destination Zone on Gen7. Dest Zone Type (Trusted/Untrusted, etc.) on Gen6.
+    , sosUserSessionType = DeviceCustomString5Label // String indicating the user session type, determined by the auth mechanism.
+    , sosUserSessionDuration = DeviceCustomString6Label // User session duration in seconds.
+    , NetworkIcmpType = FieldDeviceCustomNumber1 // ICMP Type
+    , NetworkIcmpCode = FieldDeviceCustomNumber2 // ICMP Code
+    , SrcUsername = SourceUserName
+    , ThreatOriginalConfidence = ThreatConfidence
+    , DstNatIpAddr = DestinationTranslatedAddress
+    , DstNatPortNumber = DestinationTranslatedPort
+    , SrcNatIpAddr = SourceTranslatedAddress
+    , SrcNatPortNumber = SourceTranslatedPort
+    , HttpUserAgent = RequestClientApplication
+  | extend
+    appcat = column_ifexists("appcat", "") // Application Category from App Control. Sets an empty appcat column if column doesn't exist.

```

Comment:
@jaimeesc I did this quick test and found you don't need to do column_ifexists for columns you are extracting using parse-kv. Parse-kv will create columns for you. It would be better to remove column_if_exists conditions for all columns that already exists in schema, as its unnecessary and will lower parser performance.

Reply:
Thanks for that tip. I've removed a few of the column_ifexists() lines.

Reply:
@jaimeesc change not reflecting. can you recheck?